XMonad/Actions/WorkspaceNames.hs

```diff
@@ -185,3 +196,21 @@ workspaceNamePrompt conf job = do
                                 Just i -> i
         contains completions input =
           return $ filter (Data.List.isInfixOf input) completions
+
+setTag :: (WorkspaceId -> WorkspaceId) -> WindowSpace -> WindowSpace
+setTag remap ws = ws { W.tag = remap $ W.tag ws }
+
+ewmhWorkspaceNamesLogHook' :: ((WorkspaceId -> Maybe String) -> WorkspaceId -> String)
+                           -> X ()
+ewmhWorkspaceNamesLogHook' nameWorkspace =
+  map . setTag . nameWorkspace <$> getWorkspaceNames' >>=
+      ewmhDesktopsLogHookCustom
+
+ewmhWorkspaceNamesLogHook = ewmhWorkspaceNamesLogHook' getWorkspaceNameFromTag
+
+ewmhWorkspaceNames :: XConfig a -> XConfig a
+ewmhWorkspaceNames c = c { startupHook     = startupHook c +++ ewmhDesktopsStartup
+                         , handleEventHook = handleEventHook c +++ ewmhDesktopsEventHook
+                         , logHook         = logHook c +++ ewmhWorkspaceNamesLogHook
+                         }
+  where x +++ y = mappend y x
```

Comment:
@pjones @geekosaur 
Any thoughts on what I've written above? I'm happy to proceed in whatever way you think is best.